pkg/skoop/plugin/calico.go

```diff
@@ -655,6 +655,20 @@ func (h *calicoHost) ToService(upstream *model.Link, dst model.Endpoint, protoco
 		return nil, err
 	}
 
+	for _, sourceRange := range service.Spec.LoadBalancerSourceRanges {
+		_, cidr, err := net.ParseCIDR(sourceRange)
+		if err != nil {
+			klog.Errorf("source range(%s) format invalid, %v", cidr, err)
+			continue
+		}
+		if !cidr.Contains(pkt.Src) {
```

Comment:
Multiple source ranges should be ORed together. Suspicion should be added when none  of these cidrs contains packet source.